lib/elements/dropDown.js

```diff
@@ -30,66 +34,160 @@ class DropDown extends Element {
+          let option = this.options[i];
+          if (option.text === value || option.value === value) {
+            found = true;
+            if (option.disabled) {
+              found_value[value] = { isDisabled: true };
+              disabled = true;
+              break;
+            } else {
+              let ele = selectAndDispatchEvent(this, i);
+              found_value[value] = { isAvailable: ele };
+              return true;
+            }
+          }
+        }
+        if (disabled) {
+          return false;
+        }
+        if (!found) {
+          found_value[value] = { isAvailable: false };
+          return false;
+        }
+      });
       return found_value;
     }
     const options = setNavigationOptions({});
 
     // RegExp is not serialised. So loop through dropdown text and find a match
     // before invoking via CDP API
-    let valueToSelect = value;
-    if (isRegex(value)) {
-      let options = await this.text();
-      valueToSelect = await options.split('\n').find((option) => value.exec(option));
+
+    let valuesToSelect = values;
+    if (!Array.isArray(values)) {
+      if (values && typeof values === 'object' && ('index' in values)) {
+        if (Array.isArray(values.index)) {
+          valuesToSelect = values.index;
+        } else {
+          valuesToSelect = [values.index];
+        }
+      } else if (isRegex(values)) {
+        let options = await this.text();
+        valuesToSelect = await options.split('\n').filter((option) => values.exec(option));
+      } else {
+        valuesToSelect = [values];
+      }
     }
 
     await doActionAwaitingNavigation(options, async () => {
       const { result } = await this.runtimeHandler.runtimeCallFunctionOn(selectBox, null, {
         objectId: this.get(),
-        arg: valueToSelect,
+        arg: valuesToSelect,
         returnByValue: true,
       });
       const { stack, error } = result.value;
       if (stack) {
         throw new Error(error + '\n' + stack);
+      } else if (error) {
+        throw new Error(error);
       }
-      if (result.value.isDisabled) {
-        throw new Error(`Cannot set value ${value} on a disabled field`);
-      }
-      if (!result.value.isAvailable) {
-        throw new Error(`Option ${value} not available in drop down`);
+      let disabledValues = [];
+      let unavailableValues = [];
+      Object.keys(result.value).forEach((key) => {
+        if (result.value[key].isDisabled) {
+          disabledValues.push(key);
+        }
+        if (!result.value[key].isAvailable) {
+          unavailableValues.push(key);
+        }
+      });
+
+      if (disabledValues.length > 0) {
+        throw new Error(`Cannot set value ${disabledValues.join(',')} on a disabled field`);
+      } else if (Object.keys(result.value).length === 0) {
+        if (typeof values === 'undefined') {
+          throw new Error('Option undefined not available in drop down');
```

Comment:
Can you add a test to validate this?

Reply:
The tests has already been already in dropDown.test.js. For the disabled field, following test are added :
```
 it('test select() should throw when selecting multiple values and one of the value is disabled', async () => {
      try {
        expect(await dropDown('Foods').select(['Burger', 'Pasta']));
      } catch (err) {
        expect(err.message).to.equal('Cannot set value Burger on a disabled field');
        expect(await dropDown('Foods').value()).to.equal('');
      }
    });
```

Test for undefined was already present earlier in the master branch 
```
it('should throw error for undefined values', async () => {
      try {
        expect(await dropDown('Cars').select(undefined));
      } catch (err) {
        expect(err.message).to.equal('Option undefined not available in drop down');
      }
    });
```